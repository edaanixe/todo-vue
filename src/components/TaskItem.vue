<template>
    <li :class="{ completed: completed }">
    <div class="form-check">
        <label class="form-check-label">
        <input class="checkbox" type="checkbox" :checked="completed" @click="$emit('resolved', id)" />
        {{title}} <i class="input-helper"></i></label>
    </div>
    <i class="remove fas fa-trash-alt fa-xs" 
    @click="$emit('remove', id)">
    </i>
    </li>  
</template>

<script>
import Vue from 'vue'
export default Vue.extend({
  props: {
    completed: {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      default: 'N/A'
    },
    id: {
      type: Number,
      default: 0
    },
    onRemove: {
        type: Function
    }
  }
})
</script>