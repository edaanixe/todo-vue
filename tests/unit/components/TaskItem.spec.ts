import { shallowMount } from '@vue/test-utils'
import TaskItem from '@/components/TaskItem.vue'

describe('TaskItem.vue', () => {
  it('should match element against snapshot', () => {
    const wrapper = shallowMount(TaskItem)
    expect(wrapper.element).toMatchSnapshot()
  })


  it('should render TaskItem.title prop', () => {
    const title = 'Buy rice'
    const wrapper = shallowMount(TaskItem, {
      propsData: {
        title
      }
    })

    expect(wrapper.text()).toContain(title)    
  });

  it('should render TaskItem.completed prop', () => {
    const completed = true
    const wrapper = shallowMount(TaskItem, {
      propsData: {
        completed
      }
    })

    expect(wrapper.classes()).toContain('completed')
  });  

  it('should trigger remove event', async () => {
    const wrapper = shallowMount(TaskItem)

    await wrapper.find('.remove').trigger('click')

    expect(wrapper.emitted('remove')).toHaveLength(1)
  });

  it('should trigger resolved event', async () => {
    const wrapper = shallowMount(TaskItem)

    await wrapper.find('.checkbox').trigger('click')

    expect(wrapper.emitted('resolved')).toHaveLength(1)
  });  
})
