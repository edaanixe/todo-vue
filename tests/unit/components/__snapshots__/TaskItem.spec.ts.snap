// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem.vue should match element against snapshot 1`] = `
<li
  class=""
>
  <div
    class="form-check"
  >
    <label
      class="form-check-label"
    >
      <input
        class="checkbox"
        type="checkbox"
      />
      
    N/A 
      <i
        class="input-helper"
      />
    </label>
  </div>
   
  <i
    class="remove fas fa-trash-alt fa-xs"
  />
</li>
`;
